### DISCUSSION TITLE
Channel Entries parameter Orderby category

### DISCUSSION THREAD

**Discussion by ic360-com** [ExpressionEngine/ExpressionEngine#2995] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Rather than using the Channel Categories method to output all the Entries contained within a Channel's Categories, it would be nice to have this as an option within the exp:channel:entries tag.

**Describe the solution you'd like**
{exp:channel:entries channel="vehicle" category="1|3|2" orderby="category|price" sort="asc" dynamic="no"}
By having a 'category' option for the orderby parameter, it would output all the Entries for the Categories specified, but grouped in the same sequence. eg. All the Entries for Category 1, then all the Entries for Category 3 and then all the Entries for Category 2.

**Describe alternatives you've considered**
{exp:channel:categories channel="vehicle" style="linear"}
	{exp:channel:entries channel="vehicle" category="{category_id}" orderby="price" sort="asc" dynamic="no"}
This method also works, but it does mean I've had to use a PHP script to create a {switch} function. Using 'switch' in this setup only references the local exp:channel:entries loop, and not the overall number of Entries being output to the page.

**Comment by paulcripps** [comment]
Semi related, I've often needed to order by channel name, eg. orderby="vehicle".

**Comment by robinsowell** [comment]
Ordering by channel name wouldn't be a difficult add, but the category one is tricky because an entry can be in multiple categories.  Are you thinking they just show the entry in each category, a la the category archive tag?

I do like the idea- it's a use case that does come up periodically.

**Comment by ic360-com** [comment]
Hmmm, yes I see what you mean about the multiple categories situation. Not sure how to get around that one - maybe show each Entry just the once (or have another parameter that allow Entries to be displayed for every Category it's contained in).

This is the use case I would find it useful for:
https://www.duncanselfdrive.com/car-and-van-hire
Channel = Vehicles
Categories = Vans, Minibuses, Cars

I've used the Channel Categories and the Channel Entries tags to output all the vehicles (grouped by Category) in the Daily Price order (ascending).
